ddtrace/contrib/unittest/patch.py

```diff
@@ -0,0 +1,183 @@
+    ),
+)
+
+
+def _set_tracer(tracer):
+    unittest._datadog_tracer = tracer
+
+
+def _store_span(item, span):
+    """Store span at `unittest` instance."""
+    item._datadog_span = span
+
+
+def _extract_span(item):
+    """Extract span from `unittest` instance."""
+    return getattr(item, "_datadog_span", None)
+
+
+def _extract_test_method_name(item):
+    """Extract test method name from `unittest` instance."""
+    return getattr(item, "_testMethodName", None)
+
+
+def _extract_suite_name_from_test_method(item):
+    item_type = type(item)
+    return getattr(item_type, "__name__", None)
+
+
+def _extract_class_hierarchy_name(item):
+    item_type = type(item)
+    return getattr(item_type, "__name__", None)
+
+
+def _extract_module_name_from_test_method(item):
+    return getattr(item, "__module__", None)
+
+
+def _extract_test_skip_reason(args):
+    return args[1]
+
+
+def _extract_test_file_name(item):
+    return os.path.basename(inspect.getfile(item.__class__))
+
+
+def is_unittest_support_enabled():
+    return unittest and getattr(unittest, "_datadog_patch", False) and _CIVisibility.enabled
+
+
+def _is_valid_result(instance, args):
+    return instance and type(instance) == unittest.runner.TextTestResult and args
+
+
+def patch():
+    """
+    Patch the instrumented methods from unittest
+    """
+    if (
+        not config._ci_visibility_unittest_enabled
+        or getattr(unittest, "_datadog_patch", False)
+        or _CIVisibility.enabled
+    ):
+        return
+
+    _CIVisibility.enable(config=ddtrace.config.unittest)
+
+    unittest._datadog_patch = True
+
+    _w = wrapt.wrap_function_wrapper
+
+    _w(unittest, "TextTestResult.addSuccess", add_success_test_wrapper)
+    _w(unittest, "TextTestResult.addFailure", add_failure_test_wrapper)
+    _w(unittest, "TextTestResult.addError", add_failure_test_wrapper)
+    _w(unittest, "TextTestResult.addSkip", add_skip_test_wrapper)
+
+    _w(unittest, "TestCase.run", start_test_wrapper_unittest)
+
+
+def unpatch():
+    if not getattr(unittest, "_datadog_patch", False):
+        return
+
+    _u(unittest.TextTestResult, "addSuccess")
+    _u(unittest.TextTestResult, "addFailure")
+    _u(unittest.TextTestResult, "addError")
+    _u(unittest.TextTestResult, "addSkip")
+    _u(unittest.TestCase, "run")
+
+    unittest._datadog_patch = False
+    _CIVisibility.disable()
+
+
+def _set_test_span_status(args, status):
+    test_item = args[0]
+    span = _extract_span(test_item)
+    if not span:
+        return
+    span.set_tag_str(test.STATUS, status)
+    if status == test.Status.FAIL.value:
+        exc_info = args[1]
+        span.set_exc_info(exc_info[0], exc_info[1], exc_info[2])
+    elif status == test.Status.SKIP.value:
+        span.set_tag_str(test.SKIP_REASON, _extract_test_skip_reason(args))
+
+
+def add_success_test_wrapper(func, instance, args, kwargs):
+    if is_unittest_support_enabled() and _is_valid_result(instance, args):
```

Comment:
Same concern here. We only parse the args and not the kwargs.

Reply:
I'm not sure if I follow, the `add_success_test_wrapper` function gets called whenever `unittest` calls `TextTestResult.addSuccess` which happens whenever a test passes. We only need to set the span as `PASS`. Furthermore, the `kwargs` are empty in this function call.